src/common/buffer.cc

```diff
@@ -1088,14 +1088,20 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
   buffer::list::iterator_impl<is_const>::iterator_impl(bl_t *l, unsigned o)
     : bl(l), ls(&bl->_buffers), off(0), p(ls->begin()), p_off(0)
   {
-    advance(o);
+    advance(static_cast<ssize_t>(o));
   }
 
   template<bool is_const>
   buffer::list::iterator_impl<is_const>::iterator_impl(const buffer::list::iterator& i)
     : iterator_impl<is_const>(i.bl, i.off, i.p, i.p_off) {}
 
   template<bool is_const>
+  void buffer::list::iterator_impl<is_const>::advance(int o)
+  {
+    advance(static_cast<ssize_t>(o));
+  }
+
+  template<bool is_const>
```

Comment:
@liewegas i doubt it, because when C++ mangles the method signature it takes its arguments into consideration. let's see if the test passes.